### PR TITLE
C#: Fix names of generic types/methods in model editor queries

### PR DESCRIPTION
**Pull request by koesie10**
The names returned in the C# model editor queries did not match the names that were expected in the MaD files. For example, it would return `PublicGenericClass`2`, whereas the MaD format expects `PublicGenericClass<T,T2>`. This fixes it by reusing the same `QualifiedName` module that `ExternalFlow.qll` uses.

### CODE REVIEW

File: csharp/ql/src/utils/modeleditor/ModelEditor.qll [hunk 1]
```diff
@@ -34,7 +34,17 @@ class Endpoint extends Callable {
...
-  string getTypeName() { result = nestedName(this.getDeclaringType().getUnboundDeclaration()) }
+  string getTypeName() {
+    result = qualifiedTypeName(this.getNamespace(), this.getDeclaringType().getUnboundDeclaration())
+  }
+
+  /**
+   * Gets the qualified method name of this endpoint.
+   */
+  bindingset[this]
+  string getMethodName() {
```
Comment (michaelnebel) [inline csharp/ql/src/utils/modeleditor/ModelEditor.qll:45]: Maybe consider to use `getEndpointName` or `getCallableName`. The endpoint logic also covers property accessors, which are not classified as methods.

**Review by michaelnebel — APPROVED** [review]
Looks good to me - only a minor suggestion related to naming.

File: csharp/ql/src/utils/modeleditor/ModelEditor.qll (hunk 1 above)
Comment (koesie10) [reply csharp/ql/src/utils/modeleditor/ModelEditor.qll:45]: Thanks, that makes sense. I've renamed it to `getEndpointName`.

**Review by michaelnebel — APPROVED** [review]
LGTM 👍

**Pull request merged**